cpp/src/copying/copy.cu

```diff
@@ -132,14 +131,26 @@ std::unique_ptr<column> copy_if_else( Left const& lhs,
 
    auto bool_mask_device_p = column_device_view::create(boolean_mask);
    column_device_view bool_mask_device = *bool_mask_device_p;                                    
-   
-   auto filter = [bool_mask_device] __device__ (cudf::size_type i) { return bool_mask_device.element<cudf::experimental::bool8>(i); };
-   return cudf::experimental::type_dispatcher(lhs.type(),                                             
-                                             copy_if_else_functor{},
-                                             lhs, rhs, boolean_mask.size(),
-                                             left_nullable, right_nullable,
-                                             filter,
-                                             mr, stream);
+  
```

Comment:
This works but see Jake's comment here.  https://github.com/rapidsai/cudf/issues/3856#issuecomment-576949361

Reply:
I tried it, but there were limitations due to the iterator form, so I had to switch back to this.

Reply:
:+1: